### ISSUE TITLE
nan value from weave on HIV dataset

### ISSUE THREAD

**Issue by gaojianl** [deepchem/deepchem#2946]
Hello~  
I trained Weave on the Hiv dataset, but both the loss and predicted value is nan. When I change the dataset into Bace or change the Weave into GraphConv, the bug disappeared.
It seems that some molecule are not properly prepared in the HIV, just like 'CCC[N+](CCC)(CCC)CCC.O=S(O)(=[OH+])c1cc(N=Nc2cc(S(=O)(O)=[OH+])c3cccnc3c2O)ccc1C=Cc1ccc(N=Nc2cc(S(=O)(O)=[OH+])c3cccnc3c2O)cc1S(=O)(O)=[OH+]' with salts unremoved. But removing this molecule did not make difference for that bug. Could anyone please check this?
Thanks very much!

Here's my code:
tasks_, datasets_, transformers_ = dc.molnet.load_hiv(featurizer='Weave', splitter='scaffold', reload=False)
train_dataset, valid_dataset, test_dataset = datasets
model = WeaveModel(mode='classification', n_tasks=1, batch_size=32, learning_rate=1e-1, dropout=0.05, device='cuda:0')
loss = model.fit(train_dataset)

## Environment
* Conda 4.9
* Python 3.7
* DeepChem 2.6.1

**Comment by arunppsg** [comment]
I think the molecules is not getting parsed properly. I made a small example to reproduce it:
```
from rdkit import Chem
smiles = 'CCCN+(CCC)CCC.O=S(O)(=[OH+])c1cc(N=Nc2cc(S(=O)(O)=[OH+])c3cccnc3c2O)ccc1C=Cc1ccc(N=Nc2cc(S(=O)(O)=[OH+])c3cccnc3c2O)cc1S(=O)(O)=[OH+]'
mol = Chem.MolFromSmiles(smiles)
```

The above code raises an error like `SMILES Parse Error: syntax error while parsing:`. Not sure what's happening. 

cc @rbharath

**Comment by rbharath** [comment]
I think the issue is that Weave is unstable for some hyperparameters and sometimes produces NaNs. I'd recommend playing with the learning rate to see if you can stabilize trainiing.

Typically on a large dataset you run into a few smiles parsing errors. I think that's now the root of the NaN though

**Comment by gaojianl** [comment]
> I think the issue is that Weave is unstable for some hyperparameters and sometimes produces NaNs. I'd recommend playing with the learning rate to see if you can stabilize trainiing.
> 
> Typically on a large dataset you run into a few smiles parsing errors. I think that's now the root of the NaN though

Thanks for your kind reply! 
I've tried both with smaller lr and delete molecules with solvent before, but none of them worked. There probably is a problem in the Weave featurization of HIV dataset.

**Comment by gaojianl** [comment]
> I think the molecules is not getting parsed properly. I made a small example to reproduce it:
> 
> ```
> from rdkit import Chem
> smiles = 'CCCN+(CCC)CCC.O=S(O)(=[OH+])c1cc(N=Nc2cc(S(=O)(O)=[OH+])c3cccnc3c2O)ccc1C=Cc1ccc(N=Nc2cc(S(=O)(O)=[OH+])c3cccnc3c2O)cc1S(=O)(O)=[OH+]'
> mol = Chem.MolFromSmiles(smiles)
> ```
> 
> The above code raises an error like `SMILES Parse Error: syntax error while parsing:`. Not sure what's happening.
> 
> cc @rbharath

Thank you arunppsg.  I can parse this smile correctly with rdkit==2020.9.5. But I think the solvent might be harmful for the dataset.
